doc/designs/edns.md

```diff
@@ -0,0 +1,141 @@
+
+
+| Option | Description                                       |
+|:------------------------ | :------------------------------ |
+|    --dns-over-tls | enable DNS over TLS support. This option is present on both client and server. It deploys Unbound and configures BIND on the server to receive DoT requests. |
+    --dot-forwarder | the upstream DNS server with DoT support. It must be specified in the format 1.2.3.4#dns.server.test |
+    --dns-over-tls-key and --dns-over-tls-cert | in case user prefers to have the DoT certificate in BIND generated by themselves. If these are empty, IPA CA is used instead to request a new certificate. |
+
+
+
+### Configuration
+
+Any configuration options? Any commands to enable/disable the feature or turn on/off its parts? 
+
+TBD.
+
+## Upgrade
+The following use cases must be covered by the feature implementation:
+
+### UC1: Upgrade and Enable DoT
+The implementation must handle upgrade procedure by disabling systemd-resolved in favor of Unbound, and configure Bind (named) to accept incoming encrypted queries.
+
+### UC2: Upgrade without Enabling DoT
+The implementation must handle upgrade procedure by disabling systemd-resolved in favor of Unbound, and configure Bind (named).
+
+
+## Test plan
+
+TBD. PRCI impact, FreeIPA DNS test suites impact. 
+
+### Test Cases:
+
+- TC1: Verify that the new eDNS options are visible and accessible in the ipa CLI. Test basic functionalities of the new eDNS options, such as setting and updating DNS configurations.
+- TC2: Setting up IPA Server with the --dns-over-tls option. Ensure that the server setup process completes successfully without errors related to eDNS configurations. 
+- TC3. Setting up IPA Server with the --dns-over-tls-cert option with a valid certificate. Ensure that the server setup process completes successfully without errors related to eDNS configurations
+- TC4. Setting up IPA Server with the --dns-over-tls-cert option with an invalid certificate. Ensure that errors are reported when an invalid certificate is specified. 
+- TC5. Setting up IPA replica with the --dns-over-tls option. Ensure that the server setup process completes successfully without errors related to eDNS configurations.
+- TC6. Setting up IPA replica with the --dns-over-tls-cert option with a valid certificate. Ensure that the server setup process completes successfully without errors related to eDNS configurations.
+- TC7. Setting up IPA replica with the --dns-over-tls-cert option with an invalid certificate. Ensure that errors are reported when an invalid certificate is specified.
+- TC8. Setting up IPA client with the --dns-over-tls option. Ensure that the IPA client setup process completes successfully without errors related to eDNS configurations and validate its connectivity to the server 
+- TC9. Testing DNS Traffic Encryption. Send DNS queries and analyze the traffic to confirm that DNS communication is encrypted when using the eDNS options.
```

Comment:
For this, verbosity can be increased in Unbound logs, then the individual queries will be shown and whether they're encrypted or not (using port 853).

Reply:
added